nifi-framework-bundle/nifi-framework-extensions/nifi-py4j-framework-bundle/nifi-python-framework/src/main/python/framework/ProcessorInspection.py

```diff
@@ -47,6 +47,70 @@ def visit_Assign(self, node):
                             self.string_assignments.append[variable_name] = string_value
         self.generic_visit(node)
 
+class CollectPropertyDescriptorVisitors(ast.NodeVisitor):
+
+    def __init__(self, module_string_constants):
+        self.module_string_constants = module_string_constants
+        self.discovered_property_descriptors = {}
+        self.logger = logging.getLogger("python.CollectPropertyDescriptorVisitors")
+
+    def resolve_dependencies(self, node: ast.AST):
+        resolved_dependencies = []
+        for dependency in node.elts:
+            variable_name = dependency.args[0].id
+            if not self.discovered_property_descriptors[variable_name]:
+                self.logger.error(f"Not able to find actual property descriptor for {variable_name}, so not able to resolve property dependencies.")
+            else:
+                actual_property = self.discovered_property_descriptors[variable_name]
+                dependent_values = []
+                for dependent_value in dependency.args[1:]:
+                    dependent_values.append(get_constant_values(dependent_value, self.module_string_constants))
+                resolved_dependencies.append(PropertyDependency(name = actual_property.name,
+                                                                display_name = actual_property.display_name,
+                                                                dependent_values = dependent_values))
+        return resolved_dependencies
+
+    def resolve_property_descriptor_name_in_code(self, node: ast.AST):
+        if isinstance(node.targets[0], ast.Name):
+            return node.targets[0].id
+        elif isinstance(node.targets[0], ast.Attribute):
+            return node.targets[0].attr
+        else:
+            raise Exception("Unable to determine name from source code")
+
+    def visit_Assign(self, node: ast.AST):
+        if self.assignment_is_property_descriton(node):
+            property_descriptor_name_in_code = self.resolve_property_descriptor_name_in_code(node)
+            self.logger.debug(f"Found PropertyDescriptor in the following assignment {property_descriptor_name_in_code}")
+            if not node.value.keywords:
+                self.logger.error(f"Not able to parse {property_descriptor_name_in_code} PropertyDescriptor as no keywords assignments used.")
+            else:
+                descriptor_info = {}
+                for keyword in node.value.keywords:
+                    key = keyword.arg
+                    if key == 'dependencies':
+                        self.logger.info(f"Resolving dependencies for {property_descriptor_name_in_code}.")
```

Comment:
This should be a debug log, I believe. As-is, it keeps showing up in the logs because this is run periodically. So then we end up with random log messages that don't make much sense on their own.

Reply:
Yeah totally agree, forgot to change it, after I tested the code